optimizely/project_config.py

```diff
@@ -134,11 +135,45 @@ def __init__(self, datafile, logger, error_handler):
         self.experiment_feature_map = {}
         for feature in self.feature_key_map.values():
             feature.variables = self._generate_key_map(feature.variables, 'key', entities.Variable)
-
             for exp_id in feature.experimentIds:
                 # Add this experiment in experiment-feature map.
                 self.experiment_feature_map[exp_id] = [feature.id]
 
+        # all rules(experiment rules and delivery rules) for each flag
+        self.flag_rules_map = {}
+        for flag in self.feature_flags:
+
+            experiments = []
+            if not flag['experimentIds'] == '':
+                for exp_id in flag['experimentIds']:
+                    experiments.append(self.experiment_id_map[exp_id])
+            if not flag['rolloutId'] == '':
+                rollout = self.rollout_id_map[flag['rolloutId']]
+
+                rollout_experiments = self.get_rollout_experiments(rollout)
+
+                if rollout and rollout.experiments:
+                    experiments.extend(rollout_experiments)
+
+            self.flag_rules_map[flag['key']] = experiments
+
+        # All variations for each flag
+        # Datafile does not contain a separate entity for this.
+        # We collect variations used in each rule (experiment rules and delivery rules)
+        self.flag_variations_map = {}
+
+        for flag_key, rules in self.flag_rules_map.items():
```

Comment:
can you please check if `_generate_key_map` can be used?
or something similar method you can create.

Reply:
Most appropriate was to create a separate function.